testing/test_conftest.py

```diff
@@ -15,13 +16,24 @@
 
 @pytest.fixture(scope="module", params=["global", "inpackage"])
 def basedir(request, tmpdir_factory):
+    # Do not write bytecode to make tests not fail when run randomly
+    # (ConftestImportFailure with TestConftestValueAccessGlobal).
+    orig_env = os.environ.get("PYTHONDONTWRITEBYTECODE")
```

Comment:
I suggest to create a `MonkeyPatch()` instance manually and call `undo` at the end

Reply:
Turned out it was related to sys.modules instead - used SysModulesSnapshot there now.